Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -1301,6 +1292,42 @@ public class PBXProjGenerator {
         }
     }
 
+    func getInfoPlists(for target: Target) -> [Config: String] {
+        var searchForDefaultInfoPlist: Bool = true
+        var defaultInfoPlist: String?
+
+        let values: [(Config, String)] = project.configs.compactMap { config in
+            // First, if it was defined in the specs `info` object, use that.
+            if let value = target.info?.path {
+                return (config, value)
+            }
+
+            // Otherwise attempt to find the value from the build settings
+            let buildSettings = project.getTargetBuildSettings(target: target, config: config)
+            if let value = buildSettings["INFOPLIST_FILE"] as? String {
+                return (config, value)
+            }
+
+            // If we haven't yet looked for the default info plist, try doing so
+            if searchForDefaultInfoPlist {
+                searchForDefaultInfoPlist = false
+
+                if let plistPath = getInfoPlist(target.sources) {
+                    let relative = (try? plistPath.relativePath(from: projectDirectory ?? project.basePath)) ?? plistPath
+                    defaultInfoPlist = relative.string
+                }
```

Comment:
In the original implementation, `plistPath` was assigned to the `BuildSettings` dictionary as type `Path`:

https://github.com/yonaskolb/XcodeGen/blob/05ea968f703792084cc70e5a3ddef5dfcc039f0a/Sources/XcodeGenKit/PBXProjGenerator.swift#L1178-L1180

But in other cases, it was a `String`:

https://github.com/yonaskolb/XcodeGen/blob/05ea968f703792084cc70e5a3ddef5dfcc039f0a/Sources/XcodeGenKit/PBXProjGenerator.swift#L1173

Since I'm now converting them to `String` here instead of relying on `Any` like the previous implementation, one of the tests started failing because it was casting to `Path`. But I also see other tests cast to `String`, so I'm wondering if it matters that I cast to string here?

Here is an example where the tests read the build setting as `String`:

https://github.com/yonaskolb/XcodeGen/blob/05ea968f703792084cc70e5a3ddef5dfcc039f0a/Tests/XcodeGenKitTests/ProjectGeneratorTests.swift#L1370

Here is the test failure:

> generate Info.plist: 0 is not equal to 2

https://github.com/yonaskolb/XcodeGen/blob/05ea968f703792084cc70e5a3ddef5dfcc039f0a/Tests/XcodeGenKitTests/ProjectGeneratorTests.swift#L1506-L1507

If I change line 1506 above in the tests to `as? String`, it will pass. But I just want to confirm that it's ok for me to be holding the Info.plists as `String` here? The build settings seem to convert to `String` at some point anywhere (I'm just not sure where) and given that `info.path` is already a string, I don't see this as a breaking change in behaviour but would like a second opinion 🙇

Reply:
Yep, fine to use String. I'd consider it a bug that the build settings contained a `Path`. On the extremely long todo list is changing the type from `Any` to an enum of type `BuildSettingValue` that has cases of `String` or `[BuildSettingValue]`

Reply:
Thanks for confirming, its good to know! I'll update the test to expect `String` then 🚀